agent/xds/listeners.go

```diff
@@ -227,30 +228,50 @@ func (s *ResourceGenerator) listenersFromSnapshotConnectProxy(cfgSnap *proxycfg.
 		}
 	}
 	requiresTLSInspector := false
+	requiresHTTPInspector := false
 
 	err = cfgSnap.ConnectProxy.DestinationsUpstream.ForEachKeyE(func(uid proxycfg.UpstreamID) error {
 		svcConfig, ok := cfgSnap.ConnectProxy.DestinationsUpstream.Get(uid)
 		if !ok || svcConfig == nil {
 			return nil
 		}
-
-		for _, address := range svcConfig.Destination.Addresses {
-			clusterName := clusterNameForDestination(cfgSnap, uid.Name, address, uid.NamespaceOrDefault(), uid.PartitionOrDefault())
+		configuredPorts := make(map[int]interface{})
+		if structs.IsProtocolHTTPLike(svcConfig.Protocol) {
+			if _, ok := configuredPorts[svcConfig.Destination.Port]; ok {
+				return nil
+			}
+			clusterName := clusterNameForDestination(cfgSnap, uid.Name, fmt.Sprintf("%d", svcConfig.Destination.Port), uid.NamespaceOrDefault(), uid.PartitionOrDefault())
 			filterChain, err := s.makeUpstreamFilterChain(filterChainOpts{
-				routeName:   uid.EnvoyID(),
-				clusterName: clusterName,
-				filterName:  clusterName,
-				protocol:    svcConfig.Protocol,
-				useRDS:      structs.IsProtocolHTTPLike(svcConfig.Protocol),
+				routeName:  clusterName,
```

Comment:
optional: clusterName is formatted `destination.<address>.<service>.....` Should we change the syntax for the routes since they can handle multiple addresses?

Reply:
I'm not sure about this. My understanding is that the way routes relate to filterchains is by route_name.
In that case we have a filterchain per port so a route per port and for each address for a given port we will have a mapping in the relevant route to a cluster.

routes and filterchaines are named `destination.<port>.<service>....`
clusters and routes mapping are named `destination.<address>.<service>....`

am I missing something?

Reply:
Ah, I see you are replacing the address with the port number in the name now. Cool!

Reply:
yeah but you are still right about not to use svcConfig.Name as it will be specific to the current destination